cloud/ec2_util.go

```diff
@@ -258,6 +258,22 @@ func expandUserData(userData string, expansions map[string]string) (string, erro
 	return expanded, nil
 }
 
+// 16kB
+const userDataSizeLimit = 16 * 1024
+
+func validateUserDataSize(userData, distroID string) error {
+	if len(userData) < userDataSizeLimit {
+		return nil
+	}
+	err := errors.New("user data size limit exceeded")
+	grip.Critical(message.WrapError(err, message.Fields{
```

Comment:
if a particular distro is getting close to this limit maybe this would be too noisy in the ops-alerts channel?
Maybe an alternative could be to set up a Splunk alert?

Reply:
Done, but I really want build to stop relying on setup scripts and user data in the distro settings and this is only meant as a temporary hack solution.